### PR TITLE
SAK-41810: Error with Extra point if there is a category 0%

### PR DESCRIPTION
**Pull request by raulsv**
Avoid to set 0% to a weighted category.

### CODE REVIEW

File: edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java [hunk 1]
```diff
@@ -3336,8 +3336,8 @@ public void updateGradebookSettings(final String gradebookUid, final GradebookIn
 		if (gradebook.getCategory_type() == GradebookService.CATEGORY_TYPE_WEIGHTED_CATEGORY) {
 			double totalWeight = 0;
 			for (final CategoryDefinition newDef : newCategoryDefinitions) {
-
-				if (newDef.getWeight() == null) {
+				BigDecimal bg = newDef.getWeight() == null ? null: new BigDecimal(newDef.getWeight());
+				if (bg == null || bg.compareTo(BigDecimal.ZERO) == 0) {
```
Comment (adrianfish) [inline edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java]: need a space after the colon

File: gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/pages/SettingsPage.java [hunk 2]
```diff
@@ -131,7 +131,8 @@ public void onValidate() {
 					BigDecimal totalWeight = BigDecimal.ZERO;
 					for (final CategoryDefinition cat : categories) {
 
-						if (cat.getWeight() == null) {
+						BigDecimal catWeight = (cat.getWeight() == null) ? null : new BigDecimal(cat.getWeight());
+						if (catWeight == null || catWeight.compareTo(BigDecimal.ZERO)==0) {
```
Comment (adrianfish) [inline gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/pages/SettingsPage.java]: Needs spaces around ==

File: gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/pages/SettingsPage.java [hunk 3]
```diff
@@ -229,6 +230,7 @@ public void onSubmit(final AjaxRequestTarget target, final Form f) {
 			@Override
 			public void onError(final AjaxRequestTarget target, final Form<?> form) {
 				target.add(SettingsPage.this.feedbackPanel);
+				target.appendJavaScript("scroll(0,0);");
```
Comment (adrianfish) [inline gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/pages/SettingsPage.java]: Is this to scroll to the top and see the error? If so, a comment would be good here.

Comment (raulsv) [reply gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/pages/SettingsPage.java]: Hi @adrianfish 

All changes done. Thanks!

**Pull request merged**